### PR TITLE
AADB2COidcLoginConfigurer OAuth2 Additional Parameters

### LINKED ISSUE #17751: [FEATURE REQ] AADB2COidcLoginConfigurer OAuth2 Additional Parameters
https://github.com/microsoft/azure-spring-boot/issues/938

### PR DESCRIPTION
**Pull request by jacko9et**
Fixes #17751

### CODE REVIEW

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/b2c/AADB2CAuthorizationRequestResolver.java [hunk 1]
```diff
@@ -89,12 +92,11 @@ private OAuth2AuthorizationRequest getB2CAuthorizationRequest(@Nullable OAuth2Au
 
         cleanupSecurityContextAuthentication();
 
-        final Map<String, Object> parameters = new HashMap<>(request.getAdditionalParameters());
-
-        parameters.put("p", userFlow);
-        parameters.put(PARAMETER_X_CLIENT_SKU, AAD_B2C_USER_AGENT);
+        final Map<String, Object> additional = this.properties.getAdditional();
+        this.properties.getAdditional().put("p", userFlow);
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/b2c/AADB2CAuthorizationRequestResolver.java:96]: Maybe we shouldn't directly manipulate the `additional` here.

**Review by saragluna — DISMISSED** [review]

File: sdk/spring/azure-spring-boot-starter-active-directory-b2c/README.md [hunk 2]
```diff
@@ -101,6 +101,10 @@ application to use the Spring annotations and classes to protect the web app.
            sign-up-or-sign-in: ${your-sign-up-or-in-user-flow}
            profile-edit: ${your-profile-edit-user-flow}     # optional
            password-reset: ${your-password-reset-user-flow} # optional
+         additional:
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-starter-active-directory-b2c/README.md:104]: Maybe `authenticate-additional-parameters` will make it easier to understand?

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/b2c/AADB2CAuthorizationRequestResolver.java [hunk 3]
```diff
@@ -89,12 +93,14 @@ private OAuth2AuthorizationRequest getB2CAuthorizationRequest(@Nullable OAuth2Au
 
         cleanupSecurityContextAuthentication();
 
-        final Map<String, Object> parameters = new HashMap<>(request.getAdditionalParameters());
-
-        parameters.put("p", userFlow);
-        parameters.put(PARAMETER_X_CLIENT_SKU, AAD_B2C_USER_AGENT);
+        final Map<String, Object> additional = new HashMap<>();
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/b2c/AADB2CAuthorizationRequestResolver.java:96]: additionalParameters

**Review by chenrujun — CHANGES_REQUESTED** [review]

**Review by chenrujun — APPROVED** [review]
LGTM.

**Pull request merged**